dask/dataframe/core.py

```diff
@@ -4790,7 +4790,18 @@ def maybe_shift_divisions(df, periods, freq):
 
 @wraps(pd.to_datetime)
 def to_datetime(arg, **kwargs):
-    meta = pd.Series([pd.Timestamp('2000')])
+
+    meta = kwargs.pop('meta', None)
+
+    if meta is None:
+        meta = pd.Series([pd.Timestamp('2000')])
+        # try to assign the same index name
+        # (will work if arg is DataFrame or Series)
+        try:
+            meta.index.name = arg.index.name
+        except AttributeError:
+            pass
```

Comment:
In what case will the argument not have an index and that index not have a name?

Reply:
The way it's written, you could theoretically call `dd.to_datetime` on a string (just like you would with `pandas.dataframe`, no?

Reply:
oops no you're right it has to be some form of dataframe or series for map_partitions to work anyways. Will update this to remove this check